packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -208,6 +208,11 @@ class RunCommand extends RunCommandBase {
       ..addMultiOption(FlutterOptions.kEnableExperiment,
         splitCommas: true,
         hide: true,
+      )..addOption('target-platform',
+        defaultsTo: 'default',
```

Comment:
Instead of adding `default`, make this default to `null` (no option specified). Then only add the TargetPlatform to the gradle command if it is not null